### PR TITLE
Apple Pay

### PR DESCRIPTION
**Pull request by chrissrogers**
##### Summary

- Adds `Recurly.ApplePay` class and factory

##### Implementation

```js
var pricing = recurly.Pricing();

// .. pricing calls and bindings ..

var applePay = recurly.ApplePay({
  country: 'US',
  currency: 'USD',
  label: 'My monthy service',
  pricing: pricing
});

applePay.ready(() => {
  $('#init-apple-pay').on('click', event => applePay.begin());
});

applePay.on('error', err => console.error('Apple Pay error', err));
applePay.on('token', token => console.log('Apple Pay succeeded. Token:', token.id));
```

### CODE REVIEW

**Comment by snodgrass23** [comment]
No tests for the new class?

**Comment by snodgrass23** [comment]
```
var applePay = recurly.ApplePay({
  merchantIdentifier: '[merchant-id]',
  country: 'US',
  currency: 'USD',
  label: 'My monthy service',
  pricing: pricing
});
```

Do we still need the `merchantIdentifier` passed in here?

**Comment by chrissrogers** [comment]
> No tests for the new class?

Not yet. I ripped out the ones I had in a rewrite. Tests to come soon.

**Comment by chrissrogers** [comment]
> Do we still need the merchantIdentifier passed in here?

Ah nope not required. Removed from the description

**Comment by snodgrass23** [comment]
> Ah nope not required. Removed from the description

Cool. I had just told Sanjeev a couple of days ago it wasn't needed anymore, so mostly making sure I was correct and didn't need to update him.

File: lib/recurly/apple-pay.js [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+      this.configure(options);
+    }
+  }
+
+  get session () {
+    if (this._session) return this._session;
+
+    debug('Creating new Apple Pay session');
+
+    let session = new ApplePaySession(2, {
```
Comment (snodgrass23) [inline lib/recurly/apple-pay.js]: Might be nice at some point to move this `2` to a constant somewhere so it's purpose is more obvious for future dev. I had to look it up as I forgot was it was for already.

**Pull request merged**